src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.InstallRecords.cs

```diff
@@ -0,0 +1,339 @@
+
+                            if (NuGetVersion.TryParse(manifestVersionString, out NuGetVersion manifestVersion))
+                            {
+                                var packageIdentity = new PackageIdentity(manifestID + ".Manifest-" + manifestFeatureBand, manifestVersion);
+                                string uuidName = $"{packageIdentity};{HostArchitecture}";
+                                var upgradeCode = '{' + CreateUuid(UpgradeCodeNamespaceUuid, uuidName).ToString() + '}';
+                                Log.LogMessage($"Looking for upgrade code {upgradeCode} for {uuidName}");
+                                List<string> relatedProductCodes;
+                                try
+                                {
+                                    relatedProductCodes = WindowsInstaller.FindRelatedProducts(upgradeCode.ToString()).ToList();
+                                }
+                                catch (WindowsInstallerException)
+                                {
+                                    Console.WriteLine("Error getting related products for " + upgradeCode);
+                                    throw;
+                                }
+                                DependencyProvider dependencyProvider;
+                                if (relatedProductCodes.Count == 1 &&
+                                    DetectPackage(relatedProductCodes[0], out Version installedVersion) == DetectState.Present &&
+                                    (dependencyProvider = DependencyProvider.GetFromProductCode(relatedProductCodes[0])) != null)
+                                {
+                                    var manifestRecord = new WorkloadManifestRecord();
+                                    manifestRecord.ProductCode = relatedProductCodes[0];
+                                    manifestRecord.UpgradeCode = upgradeCode;
+                                    manifestRecord.ManifestId = manifestID;
+                                    manifestRecord.ManifestVersion = manifestVersion.ToString();
+                                    manifestRecord.ManifestFeatureBand = manifestFeatureBand.ToString();
+                                    manifestRecord.ProductVersion = installedVersion;
+                                    manifestRecord.ProviderKeyName = dependencyProvider.ProviderKeyName;
+
+                                    manifestRecords.Add(manifestRecord);
+
+                                    Log.LogMessage($"Found installed manifest: {manifestRecord.ProviderKeyName}, {manifestRecord.ProductCode}");
+                                }
+                                else if (relatedProductCodes.Count > 1)
+                                {
+                                    Log.LogMessage($"Found multiple product codes for {uuidName}, which is not expected for side-by-side manifests.");
```

Comment:
For this case, I think we should consider logging the ProductCode values to assist with troubleshooting.